install/upgrade.php

```diff
@@ -191,7 +191,41 @@ protected function upgrade_2_7() {
 
 	/**
 	 * Upgrades if the previous version is < 3.4.0.
+	 * Moves strings translations from post meta to term meta _pll_strings_translations.
 	 *
+	 * @since 3.4
+	 *
+	 * @return void
+	 */
+	protected function upgrade_3_4() {
+
+		$this->migrate_locale_fallback_to_language_description();
+
+		$posts = get_posts(
+			array(
+				'post_type' => 'polylang_mo',
+				'post_status' => 'any',
+				'numberposts' => -1,
+				'nopaging' => true,
+			)
+		);
+		if ( ! is_array( $posts ) ) {
+			return;
+		}
+
+		foreach ( $posts as $post ) {
+			$meta = get_post_meta( $post->ID, '_pll_strings_translations', true );
+			if ( empty( $meta ) || ! is_array( $meta ) ) {
+				continue;
+			}
+
+			$term_id = (int) substr( $post->post_title, 12 );
+			update_term_meta( $term_id, '_pll_strings_translations', $meta );
+			wp_delete_post( $post->ID );
```

Comment:
Do you think we have to delete **all** post, even if returned post meta is empty or not an array?
Because of the previous if, the post may not deleted at all.